app/frontend/wallet/shelley-wallet.ts

```diff
@@ -1,12 +1,23 @@
 import BlockchainExplorer from './blockchain-explorer'
 import {AccountManager} from './account-manager'
 import {AccountInfo} from '../types'
-import {CryptoProviderFeatures} from './constants'
+import {CryptoProviderFeatures, MAX_ACCOUNT_INDEX} from './constants'
 
 const ShelleyWallet = ({config, cryptoProvider}) => {
   const blockchainExplorer = BlockchainExplorer(config)
 
-  const accountManager = AccountManager({config, cryptoProvider, blockchainExplorer})
+  let maxAccountIndex = MAX_ACCOUNT_INDEX
+  if (cryptoProvider.getWalletName() === 'Trezor') {
+    // hotfix because of https://github.com/vacuumlabs/trezor-firmware/issues/43
+    maxAccountIndex = 20
+  }
```

Comment:
Just an idea, if maxAccount number is something dependant on hw wallet, (or its firmware version) we could maybe change this to 
```
const maxAccountIndex = Math.min(cryptoProvider.getMaxAccountIndex(), MAX_ACCOUNT_INDEX)
```
and implement bit more complex logic in trezor crypto provider, for example
```
if (version >= <x.x.x>) return 20; else return 100;
```
and implement the function in all the crypto providers, am not sure but Ledger has also a limit for account index..right?
This might be probably overkilling it but as I said, just an idea.

Reply:
I think that would be an overkill - the max account limit provided by hw wallets should be well above what Adalite supports so the limit may as well be global and this is really a temporary hotfix until Trezor gets out. Of course, in the future we may opt to add custom limits but I wouldn't introduce abstractions we don't need as that would rather complicate testing, introducing crypto-provider specific behavior and ultimately result in a less predictable behavior for users